Framework/Indexing/inc/MantidIndexing/IndexInfo.h

```diff
@@ -91,6 +100,9 @@ class MANTID_INDEXING_DLL IndexInfo {
 
   Kernel::cow_ptr<std::vector<specnum_t>> m_spectrumNumbers;
   Kernel::cow_ptr<std::vector<std::vector<detid_t>>> m_detectorIDs;
+
+  Kernel::cow_ptr<std::vector<Beamline::SpectrumDefinition>>
```

Comment:
My concern is in allowing that sharing to be broken. I think the sharing is essential, I'm not clear on the circumstances where we would want the breaking of that sharing to happen.